python/src/open_discourse/steps/speech_content/extract.py

```diff
@@ -19,18 +19,67 @@ def main(task):
     print("Starting..")
 
     for folder_path in sorted(RAW_TXT.iterdir()):
-        process_period(folder_path)
+
+        # Get regex helpers based on folder name/term
+        open_brackets, close_brackets, prefix = get_bracket_and_prefix_from_term_number(folder_path)
+        if open_brackets is None:
+            continue  # skip if not a valid folder
+
+        president_pattern = get_president_pattern()
+
+        faction_speaker_pattern = get_faction_speaker_pattern(
+            term_number=int(folder_path.stem[-2:]),
+            folder_path=folder_path,
+            open_brackets=open_brackets,
+            close_brackets=close_brackets,
+            prefix=prefix
+        )
+
+        minister_pattern = get_minister_pattern(open_brackets, close_brackets, prefix)
+
+        process_period(folder_path, president_pattern, faction_speaker_pattern, minister_pattern)
 
     return True
 
+def get_bracket_and_prefix_from_term_number(folder_path: Path):
+    """
+    Extracts the term number from the folder name and returns
+    open_brackets, close_brackets, and prefix for regex patterns.
+    """
+    if not folder_path.is_dir():
+        return None, None, None
 
-def process_period(folder_path: Path):
-    president_pattern_str = r"(?P<position_raw>Präsident(?:in)?|Vizepräsident(?:in)?|Alterspräsident(?:in)?|Bundespräsident(?:in)?|Bundeskanzler(?:in)?)\s+(?P<name_raw>[A-ZÄÖÜß](?:[^:([}{\]\)\s]+\s?){1,5})\s?:\s?"
+    match = regex.search(r"(?<=electoral_term_pp)\d{2}", folder_path.stem)
```

Comment:
I think we have a function already for matching the term number as we often need this, look at this function: https://github.com/pmayd/open-discourse/blob/main/python/src/open_discourse/helper/utils.py